### PR TITLE
Amendments made in work form

### PR DESCRIPTION
**Pull request by actlikewill**
- This PR adds amendments made to work form
- It abstacts out some of the logic in the work amendments view, forms and template so we can reuse the code to set both sides of the Amendments model.

https://www.loom.com/share/2bbb2e87e2dc443193f79a49b8078371?sid=79f5f758-1aa2-404b-b210-8c16a58fdc9f

### CODE REVIEW

File: indigo_app/templates/indigo_api/_work_form_content.html [hunk 1]
```diff
@@ -542,11 +564,10 @@ <h5 class="card-header">{% trans 'Country and locality (advanced)' %}</h5>
     (function () {
       function refresh(e) {
         htmx.trigger("#work-list-item-{{ work.pk }}", "hx_refresh_work_list_item");
+          document.removeEventListener('hide.bs.offcanvas', refresh);
       }
 
-      document.addEventListener('hide.bs.offcanvas', () => {
-        document.removeEventListener('hide.bs.offcanvas', refresh);
-      });
+      document.addEventListener('hide.bs.offcanvas', refresh);
```
Comment (actlikewill) [inline indigo_app/templates/indigo_api/_work_form_content.html:570]: @longhotsummer I noticed that the work listing refresh was not working as expected. Was it working for you?

**Review by goose-life — APPROVED** [review]
very, very cool!

File: indigo_app/templates/indigo_api/_work_form_content.html (hunk 1 above)
Comment (longhotsummer) [reply indigo_app/templates/indigo_api/_work_form_content.html:570]: possibly not, but i don't recall.

**Review by longhotsummer — APPROVED** [review]
great!

**Pull request merged**